include/nanobind/eigen/sparse.h

```diff
@@ -0,0 +1,125 @@
+/*
+    nanobind/eigen/dense.h: type casters for dense Eigen
+    vectors and matrices
+
+    Copyright (c) 2023 Wenzel Jakob
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#pragma once
+
+#include <nanobind/tensor.h>
+#include <nanobind/eigen/dense.h>
+#include <Eigen/SparseCore>
+
+static_assert(EIGEN_VERSION_AT_LEAST(3, 3, 0),
+              "Eigen matrix support in nanobind requires Eigen >= 3.3.0");
+
+NAMESPACE_BEGIN(NB_NAMESPACE)
+
+NAMESPACE_BEGIN(detail)
+
+template <typename Scalar, int Flags, typename StorageIndex>
+using EigenMapSparseMatrix = Eigen::Map<Eigen::SparseMatrix<Scalar, Flags, StorageIndex>>;
+
+/// Caster for Eigen::SparseMatrix<T>
+template <typename T> struct type_caster<T, enable_if_t<is_eigen_sparse_v<T>>> {
+    using Scalar = typename T::Scalar;
+    using StorageIndex = typename T::StorageIndex;
+    using Index = typename T::Index;
+    static constexpr bool row_major = T::IsRowMajor;
+
+    using ScalarTensor = tensor<Scalar, shape<any>, c_contig, device::cpu>;
+    using StorageIndexTensor = tensor<StorageIndex, shape<any>, c_contig, device::cpu>;
+
+    using ScalarCaster = make_caster<ScalarTensor>;
+    using StorageIndexCaster = make_caster<StorageIndexTensor>;
+
+    NB_TYPE_CASTER(T, const_name<row_major>("scipy.sparse.csr_matrix[",
+					    "scipy.sparse.csc_matrix[")
+		   + make_caster<Scalar>::Name + const_name("]"));
```

Comment:
`make_caster<Scalar>` -> `ScalarCaster`

Reply:
But `ScalarCaster = make_caster<ScalarTensor> ≠ make_caster<Scalar>`. I think we do want `make_caster<Scalar>::Name` here.

Reply:
my bad.